lib/srv/server/azure_watcher.go

```diff
@@ -119,6 +121,7 @@ type azureInstanceFetcher struct {
 	ResourceGroup     string
 	Labels            types.Labels
 	Parameters        map[string]string
+	ClientID          string
```

Comment:
I'm probably just missing something but one thing I don't understand is how is this ClientID actually being used to pick the correct Azure identity client? In this PR I see that it's being passed to the install script and then to the join config, but where is it actually being used?

Reply:
https://github.com/gravitational/teleport/blob/492e3b36d0732933531b92a1161de7a72d8a0cc8/lib/auth/register.go#L697-L708

It's been supported by the Azure join method for a while, this PR just makes it work for discovered nodes too.

Reply:
Ok cool.